pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+    @property
+    def visibility(self):
+        return self._element._element.style.visibility
+
+    @visibility.setter
+    def visibility(self, value):
+        self._element._element.style.visibility = value
+
+    @property
+    def background(self):
+        return self._element._element.style.background
+
+    @background.setter
+    def background(self, value):
+        self._element._element.style.background = value
+
+    @property
+    def color(self):
+        return self._element._element.style.color
+
+    @color.setter
+    def color(self, value):
+        self._element._element.style.color = value
+
+    @property
+    def backgroundColor(self):
+        return self._element._element.style.backgroundColor
+
+    @backgroundColor.setter
+    def backgroundColor(self, value):
+        self._element._element.style.backgroundColor = value
+
+
+class StyleCollection:
+    def __init__(self, collection: "ElementCollection") -> None:
+        self._collection = collection
+
+    def __get__(self, obj, objtype=None):
+        return obj._get_attribute("style")
+
+    # def __set__(self, obj, value):
+    #     logging.info('Updating %r to %r', 'age', value)
+    #     obj._age = value
+    def __getitem__(self, key):
+        return self._collection._get_attribute("style")[key]
+
+    def __setitem__(self, key, value):
+        for element in self._collection._elements:
+            element.style[key] = value
+
+    def pop(self, key):
+        for element in self._collection._elements:
+            element.style.pop(key)
+
+
+class ElementCollection:
+    def __init__(self, elements: [Element]) -> None:
+        self._elements = elements
+        self.style = StyleCollection(self)
+
+    def __getitem__(self, key):
+        if isinstance(key, int):
+            return self._elements[key]
+        elif isinstance(key, slice):
+            return ElementCollection(self._elements[key])
+
+        # TODO: In this case what do we expect??
+        elements = self._element.querySelectorAll(key)
+        return ElementCollection([Element(el) for el in elements])
+
+    def _get_attribute(self, attr):
+        # As JQuery, when getting an attr, only return it for the first element
+        return getattr(self._elements[0], attr)
+
+    def _set_attribute(self, attr, value):
+        for el in self._elements:
+            setattr(el, attr, value)
+
+    @property
+    def html(self):
+        return self._get_attribute("html")
+
+    @html.setter
+    def html(self, value):
+        self._set_attribute("html", value)
+
+    # @property
+    # def style(self):
+    #     return self._get_attribute("style")
+
+    @property
+    def children(self):
+        return self._elements
+
+    def __iter__(self):
+        yield from self._elements
+
+    def __repr__(self):
+        return f"{self.__class__.__name__} (length: {len(self._elements)}) {self._elements}"
+
+
+class DomScope:
+    def __getattr__(self, __name: str) -> Any:
```

Comment:
why this isn't in the *style* class too? that's what I was talking about for properties ... we can even be smarter and convert `background_color` to `backgroundColor` JS equivalent with this orchestration.

Reply:
This is meant to provide a namespace that we can load elements from accessing their id as an `attribute` ) (for those that have an 'id' that is also a valid attribute name in Python).

The `Style` class is not meant to be a full Proxy (in that sense, now that I think, we should change the name `StyleProxy` is definitely confusing 😕 ) but a small and pythonic Style object. Users can still access the style proxy by accessing `style_instance._style`.

Reply:
> `style_instance._style`

are Python users OK with accessing "private" (`_` prefixed) fields? It's usually discouraged in the JS world due clear internal signal form authors.

Reply:
`_` is usually more of "use with caution" not forbidden. I'm not opposed to rename it to `js` or `proxy` or something else.. but I think `_...` is more correct.